madmom/ml/nn/layers.py

```diff
@@ -283,68 +364,146 @@ class LSTMLayer(Layer):
         Output gate.
     activation_fn : numpy ufunc, optional
         Activation function.
+    init : numpy array, shape (num_hiddens, ), optional
+        Initial state of the layer.
+    cell_init : numpy array, shape (num_hiddens, ), optional
+        Initial state of the cell.
 
     """
 
     def __init__(self, input_gate, forget_gate, cell, output_gate,
-                 activation_fn=tanh):
+                 activation_fn=tanh, init=None, cell_init=None):
         self.input_gate = input_gate
         self.forget_gate = forget_gate
         self.cell = cell
         self.output_gate = output_gate
         self.activation_fn = activation_fn
+        # keep the state of the layer and cell
+        if init is None:
+            init = np.zeros(self.cell.bias.size, dtype=NN_DTYPE)
+        self.init = init
+        self._prev = self.init
+        if cell_init is None:
+            cell_init = np.zeros(self.cell.bias.size, dtype=NN_DTYPE)
+        self.cell_init = cell_init
+        self._state = self.cell_init
+
+    def __setstate__(self, state):
+        # restore pickled instance attributes
+        self.__dict__.update(state)
+        # TODO: old models do not have the init attributes, thus create them
+        #       remove this initialisation code after updating the models
+        if not hasattr(self, 'init'):
+            self.init = np.zeros(self.cell.bias.size, dtype=NN_DTYPE)
+        if not hasattr(self, 'cell_init'):
+            self.cell_init = np.zeros(self.cell.bias.size, dtype=NN_DTYPE)
+        # add non-pickled attributes needed for stateful processing
+        self._prev = self.init
+        self._state = self.cell_init
+
```

Comment:
Shouldn't `LSTMLayer` also have a `__getstate__` for filtering out `self._prev` **and** `self._state` for pickling?